golang/pkg/rnr/task_nested_test.go

```diff
@@ -60,3 +62,77 @@ func TestNestedTask_GetChild(t *testing.T) {
 		t.Errorf("expecting GetChild to return nil, got %v", ct)
 	}
 }
+
+func TestNestedTask_FailFirst(t *testing.T) {
+	nt := NewNestedTask("nested task test", NestedTaskOptions{Parallelism: 1, CompleteAll: false})
+	ct1 := newMockFailingTask("child 1")
+	ct2 := newMockTask("child 2")
+
+	nt.Add(ct1)
+	nt.Add(ct2)
+	nt.SetState(pb.TaskState_RUNNING)
+
+	nt.Poll()
+
+	if ct1.pbTask.State != pb.TaskState_FAILED {
+		t.Errorf("expecting child 1 to be in failed state, got %v", ct1.pbTask.State)
+	}
+	if ct2.pbTask.State != pb.TaskState_PENDING {
+		t.Errorf("expecting child 2 to be in pending state, got %v", ct2.pbTask.State)
+	}
+	if nt.pb.State != pb.TaskState_FAILED {
+		t.Errorf("expecting child 2 to be in failed state, got %v", nt.pb.State)
```

Comment:
Is this error message correct? I think we're expecting the parent nested task in failed state, not child 2.